catalog/solr/catalog-solr-solrclient/src/main/java/ddf/catalog/source/solr/rest/SolrRest.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.codice.ddf.cxf.SecureCxfClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.internal.LinkedTreeMap;
+import com.google.gson.reflect.TypeToken;
+
+public class SolrRest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrRest.class);
+
+    private static final String SOLR_URL_PROPERTY = "solrSchemaUrl";
+
+    private static final String K1_PROPERTY = "k1";
+
+    private static final String B_PROPERTY = "b";
+
+    private LinkedTreeMap<String, Object> similarityFormat;
+
+    private List fieldTypes;
+
+    private SecureCxfClientFactory<SolrRestClient> factory;
+
+    private Float k1;
+
+    private Gson gson;
+
+    private Float b;
+
+    private String solrSchemaUrl;
+
+    public SolrRest() {
+        gson = new Gson();
+    }
+
+    public void setK1(Float k1) {
+        LOGGER.debug("{}", k1);
+        this.k1 = k1;
+    }
+
+    public void setB(Float b) {
+        LOGGER.debug("{}", b);
+        this.b = b;
+    }
+
+    public void setSolrSchemaUrl(String solrSchemaUrl) {
+        this.solrSchemaUrl = solrSchemaUrl;
+    }
+
+    public Float getK1() {
+        return k1;
+    }
+
+    public Float getB() {
+        return b;
+    }
+
+    public String getSolrSchemaUrl() {
+        return solrSchemaUrl;
+    }
+
+    public void getProperties() {
+        try {
+            String response = factory.getClient()
+                    .getFieldTypes("json");
+
+            Map<String, Object> map = gson.fromJson(response, new TypeToken<Map<String, Object>>() {
+            }
+                    .getType());
```

Comment:
Formatting seems off here.

Reply:
Oops. Looks like the code formatter got confused. Fixed.